pkg/azdiskschedulerextender/main.go

```diff
@@ -93,76 +111,136 @@ func handlePingRequest(response http.ResponseWriter, request *http.Request) {
 }
 
 func handleFilterRequest(response http.ResponseWriter, request *http.Request) {
+	var (
+		args              schedulerapi.ExtenderArgs
+		filteredNodes     []v1.Node
+		filteredNodeNames []string
+		failedNodes       map[string]string
+	)
+
+	if request.Body == nil {
+		klog.Errorf("handleFilterRequest: Error request body is empty.")
+		http.Error(response, "Error request body is empty", http.StatusBadRequest)
+		return
+	}
+
 	decoder := json.NewDecoder(request.Body)
-	defer func() {
-		if err := request.Body.Close(); err != nil {
-			klog.Errorf("handleFilterRequest: Error closing decoder: %v", err)
-		}
-	}()
 	encoder := json.NewEncoder(response)
 
-	var args schedulerapi.ExtenderArgs
 	if err := decoder.Decode(&args); err != nil {
 		klog.Errorf("handleFilterRequest: Error decoding filter request: %v", err)
 		http.Error(response, "Decode error", http.StatusBadRequest)
 		return
 	}
 
-	filteredNodes := args.Nodes.Items
+	allNodes := args.Nodes.Items
```

Comment:
It would be easier to review the change if the scheduling logic were separated from the HTTP service logic. I would recommend placing the logic in a scheduler.go file that is called from main.go. I would also recommend implementing the scheduler as an object rather than as a set of functions and global variables. This way it is easier to host the implementation in a different framework like the scheduler plug-in framework.